src/Common/src/Interop/Windows/Winsock/Interop.GetAddrInfoExW.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Net.Sockets;
+using System.Runtime.ConstrainedExecution;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+internal static partial class Interop
+{
+    internal static partial class Winsock
+    {
+        internal unsafe delegate void GetAddrInfoExCompletionCallback([In] int error, [In] int bytes, [In] NativeOverlapped* overlapped);
+
+        [DllImport(Interop.Libraries.Ws2_32, ExactSpelling = true, CharSet = CharSet.Unicode, BestFitMapping = false, ThrowOnUnmappableChar = true, SetLastError = true)]
+        internal static extern unsafe int GetAddrInfoExW(
+            [In] string name,
+            [In] string serviceName,
+            [In] int namespaceId,
+            [In] IntPtr providerId,
+            [In] ref AddressInfoEx hints,
+            [Out] out AddressInfoEx* result,
+            [In] IntPtr timeout,
+            [In] ref NativeOverlapped overlapped,
+            [In] GetAddrInfoExCompletionCallback callback,
+            [Out] out IntPtr cancelHandle
+        );
+
+        [ReliabilityContract(Consistency.WillNotCorruptState, Cer.Success)]
```

Comment:
These ReliabilityContract attributes don't have a meaning in .NET Core and can be removed.